README.md

```diff
@@ -1,111 +1,116 @@
-For example, Porter uses github.com/getporter/packages for publishing our mixin feed.
-
-[Magefile]: https://magefile.org
+# Nomad Mixin for Porter
+
+This is a [Porter](https://porter.sh) mixin for interacting with [Nomad](https://www.nomadproject.io/).
+
+## Installation
+
+todo
+
+## Mixin Syntax
+
+In your porter.yaml file, add the nomad as a mixin:
+
+```yaml
+mixins:
+  - nomad:
+      ...
+```
+
+The Mixin supports the following global settings, which will apply to all your nomad actions, unless overridden in the action
+itself:
+
+```yaml
+mixins:
+  - nomad:
+      address: ""
+      region: ""
+      namespace: ""
+      httpAuth: ""
+
+      # tls
+      caCert: ""
+      caPath: ""
+      clientCert: ""
+      clientKey: ""
+      tlsServerName: ""
+      tlsSkipVerify:
+
+      # acl
+      token: ""
+```
+
+To use the mixin in an Install/Upgrade/Uninstall step, add the "jobs" block to your porter.yaml file (using install here
+an example):
+
+```yaml
+install:
+  - nomad:
+      jobs:
+        - path: nomad/pytechco-redis.nomad.hcl
+        - path: nomad/pytechco-web.nomad.hcl
+        - path: nomad/pytechco-setup.nomad.hcl
+        - dispatch: pytechco-setup
+          meta:
+            budget: "200"
+        - path: nomad/pytechco-employee.nomad.hcl
+```
+
+The job step supports the following 3 actions which can be specified as a list item in the `jobs:` block:
+
+- `path`: runs a nomad job file using the given nomad hcl file.
+- `dispatch`: dispatches a parametrized nomad job using the given job name.
+    - `meta`: (optional): a map of key/value pairs that will be passed as metadata to the nomad job.
+    - `idPrefixTemplate` (optional): a prefix added to dispatched job IDs.
+    - `payload` (optional): base64 encoded string containng the payload to pass to the nomad job. Limited to 65536 bytes.
+- `stop`: stops a nomad job using the given job name.
+    - `purge`: (optional): if set to true, the job will be purged immediately.
+
+Each of the 3 actions above can also take in the following options, which can be passed as arguments to each individual
+nomad run, overriding any global nomad parameters set earlier:
+
+- `address` (string): the address of the nomad server to connect to.
+- `region` (string): the region of the nomad server to connect to.
+- `namespace` (string): the namespace of the nomad server to connect to.
+- `caCert` (string): the path to the CA cert file to use for TLS.
+- `caPath` (string): the path to a directory of CA cert files to use for TLS.
+- `clientCert` (string): the path to the client cert file to use for TLS.
+- `clientKey` (string): the path to the client key file to use for TLS.
+- `tlsServerName` (string): the server name to use as the SNI host when connecting via TLS.
+- `tlsSkipVerify` (bool): disables TLS host verification.
+- `token` (string): the ACL token to use when connecting to Nomad.
```

Comment:
Setting the Nomad token in the bundle would be very insecure as you'd be leaking that cluster credential to anywhere the bundle was shipped. I'm not sure how/if Porter cleans up its execution environment, but the API client you're using will automatically pick up ambient credentials from the various [Nomad environment variables](https://developer.hashicorp.com/nomad/docs/commands#environment-variables) and that seems like a much safer approach.

Longer term, you might consider having Porter opt-in to using Nomad's [OIDC](https://developer.hashicorp.com/nomad/api-docs/acl/oidc)-based login so that users only have ephemeral credentials to deal with.

(Or at the very least, recommend strongly against setting the token here in the docs.)

Reply:
This is great feedback, thank you so much for taking time to look at this! A bundle author can specify that the acl token should be a porter [credential](https://porter.sh/docs/introduction/concepts-and-components/intro-credentials/), with a template like `token: {{porter.credentials.acl_token}}` which can be passed in during runtime. This way it doesn't need to be shipped with the bundle, but instead specified when the bundle is installed/ran. I will definitely look intot he OIDC in the future. I'll also strongly advise against shipping bundles with these values hard coded.